molecule/aio/group_vars/all/molecule.yml

```diff
@@ -214,10 +214,7 @@ magnum_helm_values:
       api: 1
       conductor: 1
 magnum_image_disk_format: qcow2
-magnum_images:
-  - name: ubuntu-2204-kube-v1.27.8s
-    url: https://object-storage.public.mtl1.vexxhost.net/swift/v1/a91f106f55e64246babde7402c21b87a/magnum-capi/ubuntu-2204-kube-v1.27.8.qcow2
-    distro: ubuntu
+magnum_images: "{{ _magnum_images | last }}"
```

Comment:
This needs to be a list and then you simply have the single item in there be the the last one.

Reply:
in the magnum tasks, we upload images , need it to be an list of dict. so I update to _magnum_images[-1]
- name: Upload images
  ansible.builtin.include_role:
    name: glance_image
  loop: "{{ magnum_images }}"
  vars:
    glance_image_name: "{{ item.name }}"
    glance_image_url: "{{ item.url }}"
    glance_image_container_format: "{{ magnum_image_container_format }}"
    glance_image_disk_format: "{{ magnum_image_disk_format }}"
    glance_image_properties:
      os_distro: "{{ item.distro }}"